### DISCUSSION TITLE
Using RavenDB subscriptions as an in-memory pub/sub

### DISCUSSION THREAD

**Discussion by murshex** [ravendb/ravendb#14561] [General]
Hello,

I need a way to communicate between processes.
Using a TCP socket for this seems like a good way to shoot myself in the foot.
I can't justify implementing something like RabbitMQ or Redis pub/sub, because I don't need to communicate between processes very often.

RavenDB subscriptions seems like a great option, although one problem is that I don't need to persist the "Command" ([Example](https://ravendb.net/articles/ravendb-subscriptions-messaging-patterns)) to the database.

If I don't want to persist the Command document, is there some better solution to a problem like this?

Thanks

**Comment by ayende** [comment]
Just use them normally, and let RavenDB handle this scenario for you.
Note that a major benefit here is that this is far more easily debuggable.

**Reply by murshex** [reply]
Okay, I'm gonna give that a try. Thanks!

**Reply by georgiosd** [reply]
@c0nd3v you can always set up arbitrary queries for your subscriptions, like "from Orders where Receipt = null" or whatever

**Reply by wallaceturner** [reply]
@ayende we are using ravendb in this manner also (pub/sub) 
When the OP asked 
>If I don't want to persist the Command document
and you replied 
> let RavenDB handle this scenario for you.

how does RavenDB 'handle this scenario for you' ?

**Reply by ayende** [reply]
I meant, don't bother about skipping persistence. 
In the most common scenario, what will happen is:

* RavenDB write the new doc to the transaction journal
* The subscription picks it up
* The subscription do its work and delete it

The document never actually hit the data file, only remaining in scratch (in memory) or the transaction journal.

Those are just details you shouldn't really care about.
